inc/views/pluggable/pagination.php

```diff
@@ -143,6 +147,155 @@ public function filter_localization( $data ) {
+
+	}
+
+	/**
+	 * An array of Post types that have their own pagination control in Customizer.
+	 * 
+	 * Add post types where you give users the ability to change pagination type for archive pages.
+	 * 
+	 * @return mixed 
+	 */
+	private function post_types_with_pagination_control() {
+
+		$control_ids = array(
+			'post'    => 'neve_pagination_type',
+			'product' => 'neve_shop_pagination_type',
+		);
+
+		return apply_filters( 'neve_pagination_control_post_types', $control_ids );
+	}
+
+	/**
+	 * Filter the pagination links to decide whether or not to show the "Jump to Page" input field.
+	 * 
+	 * @param mixed $markup the pagination links from the filter.
+	 * 
+	 * @return mixed
+	 */
+	public function maybe_add_jump_to_page_input( $markup ) {
+		
+		$post_type = get_post_type();
+
+		if ( $this->jump_to_enabled( $post_type ) ) {
+			return $this->normalize_jump_to_input();    
+		}
+
+		return $markup;
+
+	}
+
+	/**
+	 * Expose our Jump to JavaScript on the page.
+	 * 
+	 * If we don't create a dedicated JS function for the inline onclick, the a tag would first scroll to the top of the page(#) and then submit the form.
+	 * 
+	 * @return void
+	 */
+	public function expose_jump_to_js() {
+
+		echo "
+		<script type='text/javascript'>
+		function nvJumpTo(e){
+			e.preventDefault();
+			document.querySelector('#nv-page-jump-form').submit()
+		}
+		</script>
+		";
+		
+	}
+
+	/**
+	 * Create jump to navigation inputs.
+	 * 
+	 * @return mixed $markup HTML to output on page.
+	 */
+	private function create_jump_to_html() {
+
+		global $wp_query;
+
+		/**
+		 * Escaping functions require args to be strings or PHPStan will throw error.
+		 */
+		$max_num_pages = esc_attr( (string) absint( $wp_query->max_num_pages ) );
+		$current_page  = ! empty( get_query_var( 'paged' ) ) ? esc_attr( (string) get_query_var( 'paged' ) ) : '';
+
+		$search_query = esc_attr( get_search_query() );
+		$search_input = ! empty( $search_query ) ? '<input id="s" type="hidden" value="' . $search_query . '" name="s" />' : '';
+		
+		/**
+		 * If plain permalinks are used we need extra inputs to handle it
+		 */
+		$permalink_structure                = get_option( 'permalink_structure' );
+		$plain_permalink_blog_archive_input = ( empty( $permalink_structure ) && $wp_query->is_posts_page ) ? '<input id="nv-page-jump-pid" type="hidden" value="' . esc_attr( $wp_query->query['page_id'] ) . '" name="page_id" />' : '';
+		$plain_permalink_cpt_archive_input  = ( empty( $permalink_structure ) && $wp_query->is_post_type_archive ) ? '<input id="nv-page-jump-cpt" type="hidden" value="' . esc_attr( $wp_query->query['post_type'] ) . '" name="post_type" />' : '';
+		$plain_permalink_cpt_taxonomy_input = ( empty( $permalink_structure ) && $wp_query->is_tax ) ? '<input id="nv-page-jump-cpt-tax" type="hidden" value="' . esc_attr( $wp_query->query_vars['term'] ) . '" name="' . esc_attr( $wp_query->query_vars['taxonomy'] ) . '" />' : '';
```

Comment:
This is not necessary, You can use https://developer.wordpress.org/reference/functions/get_pagenum_link/ to get the page link

Reply:
Hey for this can you give me more details about how I'd use the function in this case? The form is submitted to the same page and these input fields are needed for plain permalinks

If im to use that function I can get the number entered in the form from `$_GET` and do a redirect but that would mean the page redirects twice (once when the form is submitted, and second time when we do the redirect) I'm not sure what we should do here